### ISSUE TITLE
Enable samples

### ISSUE THREAD

**Issue by maririos** [Azure/azure-sdk-for-net#22875]
Currently most samples are ignored and not running in nightly tests. This is because our test infrastructure creates documents on the fly, uploads them to storage, and then we run our tests.

For samples, we want them as easier as possible for user to understand so having hidden functions are not a good idea. We need to come up with a way in which we can enable our tests and make it easy for customers to understand what are we doing.

**Comment by AhmedLeithy** [comment]
I have a few suggestions. I will present them here and then I can implement whichever one we decide to go with.

1.  Let each sample in the class "DocumentTranslationSamplesClass" have parameters according to what properties are needed (Source Container SAS URI for example), which will then be used in the code. The [Test] attributes will be removed as these are not the functions that will directly be called for testing. The customer is **not exposed to any test related functionality** regarding our testing, and should understand that the **parameters are external to the use case**. A different file with [Test] functions can make the required testing preparations for each sample use case, and call the sample function.

2. **Rely on MarkDowns for clarity**. We can create a markdown file for each sample (expanding our current collection), where we can use snippets to present the use cases without exposing testing related specifics. In this case, we would use the current sample files to test normally.

3. Have a **parallel test file for each sample file**, with the same code but with testing enabled. While this preserves the customer experience as they will be able to see clean samples, it would be **more difficult to maintain** as we would need to enforce the same behavior in each corresponding file.



Feel free to add or expand on any of these, and I will also add any other ideas that come to mind.

**Comment by AhmedLeithy** [comment]
I have a few suggestions. I will present them here and then I can implement whichever one we decide to go with.

1.  Let each sample in the class "DocumentTranslationSamplesClass" have parameters according to what properties are needed (Source Container SAS URI for example), which will then be used in the code. The [Test] attributes will be removed as these are not the functions that will directly be called for testing. The customer is **not exposed to any test related functionality** regarding our testing, and should understand that the **parameters are external to the use case**. A different file with [Test] functions can make the required testing preparations for each sample use case, and call the sample function.

2. **Rely on MarkDowns for clarity**. We can create a markdown file for each sample (expanding our current collection), where we can use snippets to present the use cases without exposing testing related specifics. In this case, we would use the current sample files to test normally.

3. Have a **parallel test file for each sample file**, with the same code but with testing enabled. While this preserves the customer experience as they will be able to see clean samples, it would be **more difficult to maintain** as we would need to enforce the same behavior in each corresponding file.



Feel free to add or expand on any of these, and I will also add any other ideas that come to mind.

**Comment by maririos** [comment]
Thanks @AhmedLeithy I like option 1 the best. We can do some modifications to it to make sure we have coverage.
The idea of running our samples as tests is to make sure the samples are always up to date and the code is runnable. We've had cases where we release something and then a user tries to run our samples and realizes they don't work because we forgot to update them.

So with that context, what we can do is ask for the container SAS URI like u are saying, and like we do for the api key and endpoint https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/tests/samples/SampleSnippets.cs#L21
so in the `#else` section, we can call our `` functions and get the respective SAS URIs. Like we do here: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/tests/TranslationOperationLiveTests.cs#L29

What do you think?

**Comment by AhmedLeithy** [comment]
So, to check that I am understanding correctly, for the [sample snippets in SampleSnippets.cs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/tests/samples/SampleSnippets.cs#L21), we can put the code for getting the SAS URIs in the `#else` section. 

And then for other sample files like [this document status here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/tests/samples/Sample_DocumentStatus.cs), we can set the SAS Uris to be parameters and set them up for testing elsewhere. Correct?

That seems pretty good. I will start working on that asap.

**Comment by maririos** [comment]
The samples snippets was just an example so you could see how we can tell Roslyn how to compile the code.
you can apply that in the other samples like https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/translation/Azure.AI.Translation.Document/tests/samples/Sample_StartTranslationAsync.cs

So something like this:
```
public async Task StartTranslationAsync()
{
            string endpoint = TestEnvironment.Endpoint;
            string apiKey = TestEnvironment.ApiKey;

            var client = new DocumentTranslationClient(new Uri(endpoint), new AzureKeyCredential(apiKey));

            #region Snippet:StartTranslationAsync
#if SNIPPET
            Uri sourceUri = new Uri("<source SAS URI>");
            Uri targetUri = new Uri("<target SAS URI>")
#else
            Uri sourceUri = await CreateSourceContainerAsync(<document>);
            Uri targetUri = await CreateTargetContainerAsync();
#endif
            ...
```


Writing this I noticed that we also need logic for the `<document>`. I think we have two options.
1. We can save a `sample.txt` file in the `samples/assets` folder and create a `TestDocument` with that content
2. We can just do: 
```
#else
        protected static readonly List<TestDocument> oneTestDocuments = new()
        {
            new TestDocument("Document1.txt", "First english test document"),
        };
        Uri sourceUri = await CreateSourceContainerAsync(oneTestDocuments);
        Uri targetUri = await CreateTargetContainerAsync();
#endif 


what do you think?
```

**Comment by AhmedLeithy** [comment]
Ah okay that makes sense. But what about sample files that do not have snippets in them? Should we still use this approach with the: 
`#If SNIPPET
...
#else
....
#endif ` 

I assumed the customers would be looking at those samples in code (not throught snippets). Would it be confusing for them to see code between preprocessor directives like these? especially if the file does not contain a snippet?

If not, then I can get started right away.

as for logic for the <document>, I think we can create the TestDocument on the fly like in option 2. Similar to how it is done in testing (for consistency).

**Comment by maririos** [comment]
>Would it be confusing for them to see code between preprocessor directives like these? especially if the file does not contain a snippet?

Great point. Maybe it will be too much. Let's try it this way, and if we hear from customers that the samples are too hard to follow then we can look for other strategies

**Comment by maririos** [comment]
Pr https://github.com/Azure/azure-sdk-for-net/pull/23072
